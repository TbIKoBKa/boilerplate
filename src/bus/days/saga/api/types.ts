// Types
import { Days } from '../../types';

export type FetchDaysContract = () => Promise<Days>
