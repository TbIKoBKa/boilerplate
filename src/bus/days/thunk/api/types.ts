// Types
import { Days } from '../../types';

export type FetchDays = () => Promise<Days>
