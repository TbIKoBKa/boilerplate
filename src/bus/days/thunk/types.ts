// Core
import { ActionReducerMapBuilder } from '@reduxjs/toolkit';

// Types
import * as types from '../types';

// Fetch
export const FETCH_WEATHER_ASYNC = 'days/FETCH_WEATHER_ASYNC';

// Contracts
export type FetchDaysAsyncContract = (builder: ActionReducerMapBuilder<types.Days>) => void;
