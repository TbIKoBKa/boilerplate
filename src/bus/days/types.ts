// Core
import { CaseReducer, PayloadAction } from '@reduxjs/toolkit';

// Types
export type WeatherType = 'sunny' | 'cloudy' | 'rainy'

// State
export type Day = {
    id: string
    rain_probability: number
    humidity: number
    day: number
    temperature: number
    type: WeatherType
}
export type Days = Day[]

// Contracts
export type SetWeatherContract = CaseReducer<Days, PayloadAction<Days>>
