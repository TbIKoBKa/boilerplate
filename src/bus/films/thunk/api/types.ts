// Types
import { Films } from '../../types';

export type FetchFilms = () => Promise<Films>
