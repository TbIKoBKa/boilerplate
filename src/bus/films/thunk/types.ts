// Fetch
export const FETCH_FILMS_ASYNC = 'FETCH_FILMS_ASYNC';
export type fetchFilmsActionAsync = {
    type: typeof FETCH_FILMS_ASYNC
}
export type FetchFilmsContract = () => fetchFilmsActionAsync
