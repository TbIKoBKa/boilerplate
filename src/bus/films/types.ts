// Core

export type Film = {
    title: string,
    episode_id: number,
    opening_crawl: string,
    director: string,
    producer: string,
    release_date: string,
    characters: Array<string>,
    planets: Array<string>,
    starships: Array<string>,
    vehicles: Array<string>,
    species: Array<string>,
    created: string,
    edited: string,
    url: string,
    [key: string]: any,
}

export type Films = Array<Film>;

// Set
export const SET_FILMS = 'SET_FILMS';
export type SetFilmsActionType = {
    type: typeof SET_FILMS,
    payload: Films,
}
export type SetFilmsContract = (payload: Films) => SetFilmsActionType

export type PeopleActionTypes =
    | SetFilmsActionType
