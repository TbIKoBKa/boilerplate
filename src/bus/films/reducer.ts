// Core
import { Reducer } from 'redux';

// Types
import * as types from './types';

const inittialState: types.Films = [];

export const filmsReducer: Reducer<types.Films, types.PeopleActionTypes> = (state = inittialState, action) => {
    switch (action.type) {
        case types.SET_FILMS:
            return action.payload;
        default:
            return state;
    }
};
