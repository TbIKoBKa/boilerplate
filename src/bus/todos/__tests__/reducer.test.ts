// Reducer
import { todosReducer as todos } from '../reducer';

// Actions
import {
    deleteTodoAction,
    setTodoAction,
    setTodosAction,
    updateTodoAction,
} from '../actions';

// Types
import { Todos } from '../types';

const testTodos: Todos = [
    {
        id:          '0',
        isCompleted: false,
        text:        'text',
    },
];

describe('todos reducer', () => {
    test('should handle delete todo action', () => {
        expect(todos(void 0, deleteTodoAction('0'))).toMatchSnapshot();
    });
    test('should handle set todo action', () => {
        expect(todos(void 0, setTodoAction(testTodos[ 0 ]))).toMatchSnapshot();
    });
    test('should handle set todos action', () => {
        expect(todos(void 0, setTodosAction(testTodos))).toMatchSnapshot();
    });
    test('should handle update todo action', () => {
        expect(todos(void 0, updateTodoAction(testTodos[ 0 ]))).toMatchSnapshot();
    });
});
