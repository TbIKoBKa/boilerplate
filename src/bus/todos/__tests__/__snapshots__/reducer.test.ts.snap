// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos reducer should handle delete todo action 1`] = `Array []`;

exports[`todos reducer should handle set todo action 1`] = `
Array [
  Object {
    "id": "0",
    "isCompleted": false,
    "text": "text",
  },
]
`;

exports[`todos reducer should handle set todos action 1`] = `
Array [
  Object {
    "id": "0",
    "isCompleted": false,
    "text": "text",
  },
]
`;

exports[`todos reducer should handle update todo action 1`] = `Array []`;
